mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -696,3 +702,64 @@ def update_artifacts_location(self, run_id, new_artifacts_location):
         with self.ManagedSessionMaker() as session:
             run = session.query(SqlRun).filter_by(run_uuid=run_id).first()
             run.artifact_uri = new_artifacts_location
+
+
+def _get_attributes_filtering_clauses(parsed):
+    clauses = []
+    for sql_statement in parsed:
+        key_type = sql_statement.get('type')
+        key_name = sql_statement.get('key')
+        value = sql_statement.get('value')
+        comparator = sql_statement.get('comparator')
+        if SearchUtils.is_attribute(key_type, comparator):
+            op = SearchUtils.filter_ops.get(comparator)
+            if op:
+                clauses.append(op(getattr(SqlRun, key_name), value))
```

Comment:
This seems a bit risky because `key_name` refers to an attribute name defined in the MLflow Search API, which is not necessarily guaranteed to match the name of the appropriate attribute (or any attribute) defined on the `SqlRun` class.

Can you introduce a method that converts these MLflow Search attribute names to SqlRun attribute names and introduce a unit test to ensure that it behaves properly for every valid MLflow search attribute? This will help avoid failures due to mismatches between search attribute names and SqlRun attribute names down the road. `SearchUtils.VALID_SEARCH_ATTRIBUTE_KEYS` maintains a list of all the search attributes, which may be useful for the conversion.

Reply:
It seems this check is already done in 
comp = cls._get_identifier(stripped_comparison[0].value, cls.VALID_SEARCH_ATTRIBUTE_KEYS)

(search_utils.py line 170)

Reply:
@jcuquemelle The check on line 170 of `search_utils.py` verifies that the attribute key supplied in the user's search expression, which is an MLflow Search attribute key, is valid. It does *not* map this attribute key to an attribute of the SQLAlchemy Model (`SqlRun`). Right now, these keys happen to be named exactly the same way, but they may not always be consistent in the future, since the SQLAlchemy models are an internal storage representation. Therefore, it would be useful to have a conversion method (effectively a no-op right now) that translates MLflow Search attribute names to SqlRun attribute names, in addition to a unit test. This will help us catch discrepancies in the future (e.g., if someone changes a `SqlRun` attribute name or adds a new search key that maps to a `SqlRun` attribute name with a slightly different string). Please let me know if this makes sense!

Reply:
@dbczumar I'm working on this review while @jcuquemelle is OOO.
I agree with your comment, but I think this would worth a dedicated PR.

Reply:
@dbczumar Got your point, I do agree. I propose a quick test in a coming review so that we can move forward. My thought is that we could design something that would encompass matching field names in all cases where we need a conversion in either direction, but as @Djailla said, I prefer this would be another PR.